### PR TITLE
Python API for save/load variables

### PR DESCRIPTION
**Pull request by reyoung**

### CODE REVIEW

File: python/paddle/v2/framework/save_load.py [hunk 1]
```diff
@@ -0,0 +1,143 @@
+import os
```
Comment (wangkuiyi) [inline python/paddle/v2/framework/save_load.py]: save_load => model or io ?

File: python/paddle/v2/framework/framework.py [hunk 2]
```diff
@@ -486,6 +486,11 @@ def sync_with_cpp(self):
         for block in self.blocks:
             block.sync_with_cpp()
 
+    def iter_all_vars(self):
```
Comment (wangkuiyi) [inline python/paddle/v2/framework/framework.py]: iter_all_vars => list_all_vars

File: python/paddle/v2/framework/tests/test_fit_a_line.py [hunk 3]
```diff
@@ -4,6 +4,7 @@
 import paddle.v2.framework.optimizer as optimizer
 
 from paddle.v2.framework.framework import Program, g_program
+from paddle.v2.framework.save_load import save_persistables, load_persistables
```
Comment (wangkuiyi) [inline python/paddle/v2/framework/tests/test_fit_a_line.py]: It seems from this example that what users need is save/load_persistables. Then, do we need save/load_params?

Comment (reyoung) [reply python/paddle/v2/framework/tests/test_fit_a_line.py]: Yes, save/load_params used for inference.

**Pull request merged**